ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -181,9 +182,10 @@ txSubmissionInbound
   -> NumTxIdsToAck  -- ^ Maximum number of unacknowledged txids allowed
   -> TxSubmissionMempoolReader txid tx idx m
   -> TxSubmissionMempoolWriter txid tx idx m
+  -> (tx -> SizeInBytes) -- ^ get size of CBOR encoded transaction
   -> NodeToNodeVersion
   -> TxSubmissionServerPipelined txid tx m ()
-txSubmissionInbound tracer (NumTxIdsToAck maxUnacked) mpReader mpWriter _version =
+txSubmissionInbound tracer (NumTxIdsToAck maxUnacked) mpReader mpWriter _txSize _version =
```

Comment:
Can we verify the tx sizes using an `assert`-ion? I'd expect that production nodes run without assertions, and this would give us a chance to verify that this works well.

Reply:
Good idea - @karknu commented on slack I think that this maybe should be in an #ifdef.

Reply:
I don't think we need CPP for it, `assert`-tions are only enforced if one compiled the node with `-fno-ignore-asserts` flag.